bugnote_view_inc.php

```diff
@@ -122,6 +122,10 @@
 	$t_can_delete_all_bugnotes = access_has_bug_level( config_get( 'delete_bugnote_threshold' ), $f_bug_id );
 	$t_can_change_view_state_all_bugnotes = $t_can_edit_all_bugnotes && access_has_bug_level( config_get( 'change_view_status_threshold' ), $f_bug_id );
 
+	# Pre-generate form security tokens to avoid performance issues
+	$t_security_token_state = form_security_token( 'bugnote_set_view_state' );
```

Comment:
You mean: if no buttons are going to be shown (eg, user cant delete), no token is needed
That can be done using IF $t_can_delete... THEN generate-token-for-delete-page.
Nitpicking, but simple enough. I agree on doing it.